Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp

```diff
@@ -799,6 +802,26 @@ WebKitWebViewAccessible* View::accessible() const
 }
 #endif
 
+#if ENABLE(WPE_PLATFORM)
+void View::handleGesture(WPEGesture gesture, WPEGestureController& controller)
+{
+    // Some gestures require immediate feedback, so they are detected and handled right away.
```

Comment:
Here we would check if the view has a controller and return early otherwise.

Reply:
Then we call `wpe_gesture_controller_handle_event()` and then query the controller depending on the pahse we are (down, move, up, cancel)